staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/v2/api.proto

```diff
@@ -29,8 +29,13 @@ message EncryptedObject {
   string keyID = 2;
 
   // EncryptedDEK is the encrypted DEK.
+  // Mutually exclusive with encryptedSeed.
   bytes encryptedDEK = 3;
```

Comment:
can we add a DEK type instead of adding a seed field?

```
enum Type {
   AES_GCM = 1;
   XAES_GCM_HKDF_SHA256 = 2;
}
```

Reply:
Agree on an enum to distinguish the meaning of the field rather than a separate field containing encrypted

If we want to rename the `encryptedDEK` field while keeping the same proto tag, that's fine and wire-compatible. I like the `encrypted` prefix, and I like including `DEK` somewhere in the name since the contents of this are used as the DEK or to derive the DEK. Bad ideas that will maybe spawn good ideas:
* `encryptedDEKInput`
* `encryptedDEKSource`
* `encryptedDEK___`

Name the enum field something that matches the `encryptedDEK` field name, e.g. `encryptedDEKType`

For the enum values, we could include the details of what the contents are (e.g., for the KDF use, it's a seed):

```
AES_GCM_...
HKDF_SHA256_XNONCE_AES_GCM_SEED
```